torch/onnx/utils.py

```diff
@@ -125,30 +126,34 @@ def select_model_mode_for_export(model, mode: _C_onnx.TrainingMode):
 
 
 @contextlib.contextmanager
-def disable_apex_o2_state_dict_hook(model):
+def disable_apex_o2_state_dict_hook(model: torch.nn.Module):
     # Apex O2 hook state_dict to return fp16 weights as fp32.
     # Exporter cannot identify them as same tensors.
     # Since this hook is only used by optimizer, it is safe to
     # remove this hook while exporting.
     if not isinstance(model, torch.jit.ScriptFunction):
```

Comment:
Is this line is in conflict with the type annotation? I think if it's an nn.Module then it can't be a ScriptFunction?

Reply:
Updated to `Union[torch.nn.Module, torch.jit.ScriptFunction]`. does that seem right?